searx/preferences.py

```diff
@@ -162,9 +162,7 @@ class SearchLanguageSetting(EnumStringSetting):
     """Available choices may change, so user's value may not be in choices anymore"""
 
     def _validate_selection(self, selection):
-        if selection != "" and not match_language(
-                # pylint: disable=no-member
-                selection, self.choices, fallback=None):
+        if selection != '' and not VALID_LANGUAGE_CODE.match(selection):
```

Comment:
The first validation is not really necessary anymore because the regex should fail with an empty string. But it's also fine if you want to keep it just to be extra sure.

Reply:
In the default configuration `selection == ''`:
https://github.com/searx/searx/blob/ecb9f28869f081dc75182053c727523ed7e20d12/searx/settings.yml#L8
and
https://github.com/searx/searx/blob/ecb9f28869f081dc75182053c727523ed7e20d12/searx/preferences.py#L323-L326

So 
* `make test` fails (test_webadapter.py fails)
* the web server crash with:
```
  File "/home/alexandre/code/searx/local/py3/lib/python3.8/site-packages/flask/app.py", line 2242, in preprocess_request
    rv = func()
  File "/home/alexandre/code/searx/searx/webapp.py", line 427, in pre_request
    preferences = Preferences(themes, list(categories.keys()), engines, plugins)
  File "/home/alexandre/code/searx/searx/preferences.py", line 322, in __init__
    'language': SearchLanguageSetting(
  File "/home/alexandre/code/searx/searx/preferences.py", line 44, in __init__
    self._post_init()
  File "/home/alexandre/code/searx/searx/preferences.py", line 80, in _post_init
    self._validate_selection(self.value)
  File "/home/alexandre/code/searx/searx/preferences.py", line 166, in _validate_selection
    raise ValidationException('Invalid language code: "{0}"'.format(selection))
searx.preferences.ValidationException: Invalid language code: ""
```

In the webapp there is a fallback:
https://github.com/searx/searx/blob/ecb9f28869f081dc75182053c727523ed7e20d12/searx/webapp.py#L454-L455

And _get_browser_or_settings_language falls back to `en` (on the first and the last lines):
https://github.com/searx/searx/blob/ecb9f28869f081dc75182053c727523ed7e20d12/searx/webapp.py#L157-L163

So.... in this commit I would keep the `selection != ''`, but another PR can:
* Preferences takes the default value for locale and language.
* webapp.pre_request
    * get the defaut values for locale and language from the HTTP headers
    * fallback to `en`
    * initialize the Preferences object with these default values.
* webapp.get_locale relies only on `request.preferences.get_value('locale')`

Reply:
Yeah, you're right. My mistake.